From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Benedikt Meurer <bmeurer@chromium.org>
Date: Tue, 12 May 2026 13:23:36 +0200
Subject: [M149] [profiler] Fix race condition in code entry refcounting.

Original change's description:
> [profiler] Fix race condition in code entry refcounting.
>
> Fixed: 511290038
> Change-Id: I16f1d52578fa8d85d2552ee130f468efbae44e0e
> Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/7837982
> Reviewed-by: Philip Pfaffe <pfaffe@chromium.org>
> Auto-Submit: Benedikt Meurer <bmeurer@chromium.org>
> Commit-Queue: Benedikt Meurer <bmeurer@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#107260}

(cherry picked from commit 5472defe56a3100383935e051f87cf2eb338169d)

Bug: 514589079,511290038
Change-Id: I16f1d52578fa8d85d2552ee130f468efbae44e0e
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/7860940
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Reviewed-by: Benedikt Meurer <bmeurer@chromium.org>
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/14.9@{#14}
Cr-Branched-From: 8f08364a351ad38a60421137a09ef23953ecdd56-refs/heads/14.9.207@{#1}
Cr-Branched-From: 8de67b11924d5e8c0032029165a52d800cf05f1f-refs/heads/main@{#106999}

diff --git a/src/profiler/profile-generator.h b/src/profiler/profile-generator.h
index 38b3c713de6f3b5318cf7b6bd96c62161118d885..5834520ba29546c02aa0c0407b744b32aea9f59a 100644
--- a/src/profiler/profile-generator.h
+++ b/src/profiler/profile-generator.h
@@ -230,15 +230,13 @@ class CodeEntry {
   size_t AddRef() {
     DCHECK(is_ref_counted());
     DCHECK_LT(ref_count_, std::numeric_limits<size_t>::max());
-    ref_count_++;
-    return ref_count_;
+    return ++ref_count_;
   }
 
   size_t DecRef() {
     DCHECK(is_ref_counted());
     DCHECK_GT(ref_count_, 0UL);
-    ref_count_--;
-    return ref_count_;
+    return --ref_count_;
   }
 
   using EventField = base::BitField<LogEventListener::Event, 0, 4>;
